### ISSUE TITLE
Enabling more divergence scale options than just the two magic ones: time and divergence

### ISSUE THREAD

**Issue by corneliusroemer** [nextstrain/auspice#1769]
### Context

For mpox, it would be great to be able to switch between APOBEC divergence and non-APOBEC divergence - but as far as I know Auspice only supports two types of x-scales: time and divergence. We also only allow one of mutation count or branch length for divergence.

### Description

Allow more types of divergence scales to be added to auspice.json and picked similarly to the current time/divergence buttons.

As long as we're not studying life in other galaxies, a single time scale should be fine, but allowing multiple divergence scales might be neat!

**Comment by jameshadfield** [comment]
Yeah, this would be neat. I've thought about it in the past in the context of AA divergence. Shouldn't be _hard_, but there'll be a lot of changes as the code assumes that the tree metric is an enum `{num_date, div}`.

**Comment by corneliusroemer** [comment]
Hah, yes! I think I've also thought about it in the context of AA divergence, potentially by gene even! That'd be super neat.

If the code was typescript that might not be so hard to do but untyped JS, oof :)

**Comment by joverlee521** [comment]
This was brought up again by @victorlin in [Slack](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1774890203130689?thread_ts=1774890155.874469&cid=C01LCTT7JNN). 

This work will likely depend on https://github.com/nextstrain/auspice/issues/1238.
